### PR TITLE
Sort themes alphabetically by label

### PR DESCRIPTION
**Pull request by tahb**

### CODE REVIEW

File: lib/dfid-transition/patch/specialist_publisher/themes.rb [hunk 1]
```diff
@@ -17,22 +17,35 @@ class Themes < Base
...
+          RDF::Repository.load('http://r4d.dfid.gov.uk/RDF/SKOS/Themes.rdf')
+        end
+
+        def sorted_r4d_solutions
+          r4d_solutions.sort_by { |result| result['prefLabel'].value }
+        end
+
+        def r4d_solutions
+          sparql.query(QUERY)
+        end
```
Comment (rgarner) [inline lib/dfid-transition/patch/specialist_publisher/themes.rb:36]: There are a few issues with 'methodising' everything like this:
- The return values of `sparql` and `repository` aren't memoized, so if they are referred to twice, they'd be re-instantiated. It's a reasonable assumption for someone modifying this class in future to make that if they refer to `repository` twice, the second time they refer to it it'll be the same instance (as it is here, we'd load the triples from disk as many times as we referred to `repository`). We'd need `@repository ||= ...` to memoize here.
- The method name `sparql` is a bit misleading – it isn't a bit of SPARQL, it's a `sparql_client`. This confusion is compounded by the fact we've got a `SPARQL` constant that _does_ have SPARQL in it.
- Spreading these values out over the file makes things arguably less readable than the original one-shot method
- We pass `r4d_solutions` as a parameter to `transform_to_label_value` when we have a method of the same name in the class.
- We're querying the triples then sorting the solutions in Ruby; we could get the triplestore to do this for us by adding an `ORDER BY ?prefLabel` clause to the SPARQL.

File: spec/lib/dfid/patch/specialist_publisher/themes_spec.rb [hunk 2]
```diff
@@ -38,5 +38,14 @@
         'value' => 'Neglected%20Tropical%20Diseases'
       )
     end
+
+    it 'sorts themes alphabetically by label' do
+      patch.run
```
Comment (rgarner) [inline spec/lib/dfid/patch/specialist_publisher/themes_spec.rb]: We're not `expect`ing `RDF::Repository` to receive `load` any more here (as in the test above on #L27). That means the test will be dependent on the internet for its triples.

File: spec/lib/dfid/patch/specialist_publisher/themes_spec.rb [hunk 3]
```diff
@@ -38,5 +38,14 @@
...
+
+      schema = JSON.parse(File.read(patch_location))
+      themes = schema['facets'].find { |facet| facet['key'] == 'theme' }
```
Comment (rgarner) [inline spec/lib/dfid/patch/specialist_publisher/themes_spec.rb]: These two lines are the same as the example immediately preceding. I know how you feel about `let`, but moving them out will make both examples less noisy.

**Comment by tahb** [comment]
@rgarner I've made some updates following our call discussing your comments. Ready for another look.

**Pull request merged**